Dockerfile

```diff
@@ -95,16 +95,30 @@ RUN if [ "${DEBUG}" -gt 0 ]; then pip install --no-cache-dir -e ".[debug,xrootd]
 # hadolint ignore=DL3008,DL3013
 RUN apt-get update -y && \
     apt-get install -y --no-install-recommends \
-      git && \
+        cmake \
+        g++ \
+        gcc \
+        git \
+        libssl-dev \
+        make \
+        python3.12-dev \
+        uuid-dev && \
     if test -e modules/reana-commons; then \
       if [ "${DEBUG}" -gt 0 ]; then \
-        pip install --no-cache-dir -e "modules/reana-commons[snakemake_reports]" --upgrade; \
+        pip install --no-cache-dir -e "modules/reana-commons[snakemake, snakemake-xrootd]" --upgrade; \
       else \
-        pip install --no-cache-dir "modules/reana-commons[snakemake_reports]" --upgrade; \
+        pip install --no-cache-dir "modules/reana-commons[snakemake-xrootd]" --upgrade; \
```

Comment:
The else clause also needs to have  `"modules/reana-commons[snakemake,snakemake-xrootd]"`, otherwise Snakemake v7 from the officially released r-commons won't be overwritten by Snakemake v8 from locally-checked out r-commons branch.

Reply:
BTW since this commit will upgrade Snakemake from v7 to v8, the more appropriate commitlint scope would be something like:

```
feat(executor): upgrade to Snakemake v8 
```